### PR TITLE
generalize link_statements plugin to use "document" metadata keys

### LINKED ISSUE #740: support for generic documents (not only statements) associated to transactions
I'm looking into how Fava support documents referenced from the journal, and it is not clear to me if there is a desire to support generic documents, as opposed to only bank-style statements.

In particular, I'm interested in the ability to accept "receipts", "invoices", etc., that will generally be associated to Expenses transactions, or Assets:Receivable, etc. Is that something that Fava wants to support?

If so, I think it would be nice to rename the link_statement plugin into something more generic, and not use `#statement` as a tag, but maybe something like `#document`. I suspect that the name comes from the `document` directive of Beancount itself which, on the other hand, really seems to be statement-specific, as documents are conceptually inserted by Beancount *in between* transactions, rather than *on* them.

### PR DESCRIPTION
**Pull request by zacchiro**
Close #740

### CODE REVIEW

**Comment by zacchiro** [comment]
OK, here's the PR addressing what we discussed in #740.

Using "document" as metadata key prefix causes some naming clashes in the code (and documentation), so I've picked "disk_doc" to differentiate documents expected to be on disk and document (as Beancount entries).

Also, I've renamed the plugin to link_documents.py, the idea being that people might easily keep around the old link_statements.py file if they want to transition more gracefully. YMMV.

**Comment by zacchiro** [comment]
Oh, also, I'm not entirely sure you want to have the #document tag for documents linked by this plugin (a document tagged document sounds weird...), but I haven't found convincing ideas for a better name. Maybe "linked_document"? Ideas welcome.

File: fava/plugins/link_documents.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-"""Beancount plugin to link statements to documents.
+"""Beancount plugin to link documents on disk to document entries.
```
Comment (yagebu) [inline fava/plugins/link_documents.py]: It rather links transactions to documents, doesn't it?

**Comment by yagebu** [comment]
The code in the journal template and some other places also needs to be changed.

> Maybe "linked_document"? Ideas welcome.

Maybe just "linked"?

**Comment by yagebu** [comment]
@zacchiro: ping

**Comment by zacchiro** [comment]
@yagebu thanks for the ping, I was on VAC and this PR slipped my mind.

I've just updated it, fixing the description, using `#linked` as you suggested, and fixing all other places where I could find mention of the old `#statement` tag. If you can spot other places where adaptation is needed, please shout.

**Comment by yagebu** [comment]
Thanks. Merged without the changes to the translations, as they are maintained on POEditor.com

**Comment by vloris** [comment]
I'm having somee problems with the current fava release (1.9):

- The help-page still shows link_statements as plugin-name
- If you drag-and-drop documents on a transaction, it creates metadata named 'statement', but the new link_documents plugin doesn't look at 'statement', but at 'document' metadata.

So, I think there are still two changes necessary: update the documentation and change the metadata-key used by the statement-upload process. Does that sound correct?

**Comment by zacchiro** [comment]
@vloris thanks for the info. The first issue you mention should be fixed by #858 .
I'm not sure about the second, as I don't use the drag&drop UI, but it does sound right. I'm not the fava maintainer, but I suspect a PR fixing that would be welcome too.

**Pull request closed without merging**